Makefile

```diff
@@ -1,21 +1,94 @@
-WOKWI_PROJECT_ID=334445762078310996
-# logic puzzle and muxes
-# 4 inverters 334348818476696146
-# the clock divider 334335179919196756
+# Set the project ID for Wokwi, some examples:
+#  - 334445762078310996: Logic puzzle and muxes
+#  - 334348818476696146: Four inverters
+#  - 334335179919196756: Clock divider
+#
+WOKWI_PROJECT_ID = 334445762078310996
+
+# ======== In most cases, changes don't need to be made below this line ========
+
+# ==============================================================================
+# Flow Setup
+# ==============================================================================
+
+# NOTE: GDS & lint stages require 'PDK_ROOT' to be set
+
+SRC_DIR        ?= src
+USER_MOD_PATH  ?= $(SRC_DIR)/user_module_$(WOKWI_PROJECT_ID).v
+SCAN_WRAP_PATH ?= $(SRC_DIR)/scan_wrapper_$(WOKWI_PROJECT_ID).v
+CFG_TCL_PATH   ?= $(SRC_DIR)/config.tcl
+PROJ_ID_PATH   ?= $(SRC_DIR)/ID
+
+# Verilator (for lint)
+VERILATOR_TAG        ?= 4.106
+VERILATOR_IMAGE_NAME ?= verilator/verilator:$(LINT_DOCKER_TAG)
```

Comment:
where does the verilator docker come from?
also, how much time does this linting stage cost?

As the default wokwi use case is all auto-generated verilog I think we only need to lint the cells.v and standard output from wokwi once. This would be very useful in the verilog/HLS specific repo.

Reply:
> where does the verilator docker come from?

This appears to be an official release stream from the Verilator project: https://hub.docker.com/u/verilator

> also, how much time does this linting stage cost?

~20 seconds according to my runs on GitHub - 99% of that is just pulling the docker image, linting itself is lightning fast on a design this small.

> As the default wokwi use case is all auto-generated verilog I think we only need to lint the cells.v and standard output from wokwi once. This would be very useful in the verilog/HLS specific repo.

While I see where you're coming from, Verilator still needs to digest the `cells.v` file to be able to elaborate and hence lint the design - else it wouldn't have a definition of these primitive modules. There are no lint issues raised by `cells.v`, so unless a user modifies it there will be no warning/error messages.

Reply:
Thanks for your answers. Regarding the last point I'm not sure I explained myself.

I'm assuming we will have a separate HDL repo for people doing Verilog or other HDL.

So this repo will only be used by wokwi schematic entry, so is there any point in linting auto generated code?

Reply:
Ah I see, I'll admit I wasn't clear that there was going to be a separate template repo for pure Verilog designs. 

That said, this check is pretty cheap and gives us a higher degree of certainty that the design will integrate into the ASIC without any problems. 

If the concern is lengthening the runtime of the GitHub action, then we could run this step in parallel with the GDS generation rather than as a precondition.

Reply:
good idea to run in parallel.